src/auto-router/auto-router.ts

```diff
@@ -4,20 +4,26 @@ import { BigNumber, BigNumberish, constants } from 'ethers'
 import { Zero0xQuoteOptions, ZeroExAggregator } from '../0x-aggregator'
 import { Erc20, StandardPermitArguments } from '../erc-20'
 import { InsufficientAllowanceError } from '../errors'
-import { IndexRouter } from '../index-router'
+import { IndexWithdrawRouter } from '../index-router'
+import {
+  defaultIndexDepositRouterAddress,
+  IndexDepositRouter,
+} from '../index-router/index-deposit-router'
 import { getDefaultPriceOracle } from '../price-oracle'
 import { getDefaultIndexHelper, Index } from '../products/index'
 import { Router } from '../router'
+import { IReserveRouter } from '../typechain/IndexDepositRouter'
 import { Address, ChainId, ChainIds, TokenSymbol } from '../types'
 
-const baseMintGas = 260_000
-const additionalMintGasPerAsset = (network: ChainId): number => {
+const UQ112 = BigNumber.from(2).pow(112)
+const baseDepositGas = 95_000
+const additionalSwapDepositGas = (network: ChainId) => {
   const gas = {
-    [ChainIds.Mainnet]: 148_000,
-    [ChainIds.CChain]: 105_000,
+    [ChainIds.Mainnet]: 115_000,
+    [ChainIds.CChain]: 100_000,
   }[network]
 
-  return gas || 135_000
+  return gas || 125_000
 }
```

Comment:
@jn-lp @Gulikovskiy  Not sure that this logic correct. What do this numbers mean now?